doc/Language/syntax.pod6

```diff
@@ -223,109 +223,193 @@ say 'code again';
+C<_>. Alphanumeric characters additionally include Unicode characters in the
+general category I<Number, Decimal Digit> (Nd).
 
 =begin code :skip-test
-# valid identifiers:
+# valid ordinary identifiers:
 x
 something-longer
 with-numbers1234
-don't
-fish-food
-π
+don't-do-that
+piece_of_π
+駱駝道                 # "Rakuda-dō", Japanese for "Way of the camel"
 =end code
 
 =begin code :skip-test
-# not valid identifiers:
-with-numbers1234-5
-42
-is-prime?
-fish--food
+# invalid ordinary identifiers:
+42                    # Identifier does not start with alphabetic character
+with-numbers1234-5    # Embedded hyphen not followed by alphabetic character
+is-prime?             # Question mark is not alphanumeric
+x²                    # Superscript 2 has Unicode general category No
 =end code
 
-You use identifiers to name of constants, types (including classes and
-modules) and routines (subs and methods); they also appear in variable
-names usually proceeded by a sigil; see L<variables|/language/variables>
-for more details.
+In the C<x²> example above, the Unicode superscript numeral C<²> is not, and
+cannot be, a part of an ordinary identifier. Still, it may form part of a valid
+expression as an operator. The expression C<$x²>, for instance, produces the
+square of variable C<$x>. This should be somewhat surprising at this point,
+since operators are L<Subs|/type/Sub>, and, as mentioned above, Subs are named using
+identifiers. So how does the superscript numeral get accepted as (part of) an
+identifier? The key to this riddle lies in the concept of I<extended
+identifiers>, which is elaborated upon in the next section.
+
+=head3 Extended identifiers
+
+It is often convenient to have names that contain arbitrary characters.
+Use cases include situations where a set of entities shares a common "short"
+name, but still needs for each of its elements to be identifiable individually.
+For example, you might use a module whose short name is C<ThatModule>, but the
+complete long name of a module includes its version, naming authority, and
+perhaps even its source language. Similarly, sets of operators work together in
+various syntactic categories with names like C<prefix>, C<infix>, C<postfix>,
+etc. The (long) names of these operators, however, often contain characters that
+are excluded from ordinary identifiers.
+
+For all such uses, you can append one or more
+L<adverbial pairs|/language/syntax#Adverbial_pairs_(colon_pairs)> to an ordinary
+identifier to create a so-called I<extended identifier>. When appended
+to an identifier (that is, in postfix position), the adverbial pair syntax
+generates unique variants of that identifier.
+
+The generic form of an adverbial pair or "colon pair" is C<:key<value>> :
+it starts with a single colon C<:>, followed by an ordinary identifier C<key>,
+in turn followed by some quoting bracketing construct, such as C«< >», C<« »> or
+C<[' ']>, which quotes one or more arbitrary characters C<value>:
 
```

Comment:
This is really unnecesary, since, as said above, it's not an adverbial pair, but an adverbial form.

Reply:
"Adverbial pair" terminology removed.